server/app/acs_enterprise/lambdas/scheduler_lambda.py

```
@@ -105,38 +115,243 @@ async def _track_timed_out_calls(self, db: AsyncSession):
+
+            # Batch query: load all configs in single query using IN clause
+            config_query = select(ScheduleConfig).where(
+                ScheduleConfig.rooftop_id.in_(rooftop_ids)
+            )
+            result = await db.execute(config_query)
+            configs = result.scalars().all()
+
+            # Create dict mapping rooftop_id -> config
+            configs_by_rooftop = {c.rooftop_id: c for c in configs}
+
+            return configs_by_rooftop
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)